File: ScorpioProto/CSharp/Example/src/DataSpawn.cs
//本文件为自动生成，请不要手动修改
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using ScorpioProto.Commons;
using ScorpioProto.Table;

namespace Datas {
public partial class DataSpawn : IData {
    
    private int _ID;
    /* <summary> 测试ID 此值必须唯一 而且必须为int型  默认值() </summary> */
    public int getID() { return _ID; }
    public int ID() { return _ID; }
    private int _TestInt;
    /* <summary> int类型  默认值() </summary> */
    public int getTestInt() { return _TestInt; }
    private string _TestString;
    /* <summary> string类型  默认值() </summary> */
    public string getTestString() { return _TestString; }
    private string _TestLanguage;
    /* <summary> 测试多国语言  默认值() </summary> */
    public string getTestLanguage() { return _TestLanguage; }
    private bool _TestBool;
    /* <summary> bool类型  默认值() </summary> */
    public bool getTestBool() { return _TestBool; }
    private Int2 _TestInt2;
    /* <summary> 自定义类型 根据ExcelConfig下 table.sco文件定义的Int2解析 类型为table_后面的名字 格式为 , 隔开  默认值() </summary> */
    public Int2 getTestInt2() { return _TestInt2; }
    private TestEnum _TestEnumName;
    /* <summary> 自定义枚举  默认值() </summary> */
    public TestEnum getTestEnumName() { return _TestEnumName; }
    
    public object GetData(string key) {
        if ("ID".Equals(key)) return _ID;
        if ("TestInt".Equals(key)) return _TestInt;
        if ("TestString".Equals(key)) return _TestString;
        if ("TestLanguage".Equals(key)) return _TestLanguage;
        if ("TestBool".Equals(key)) return _TestBool;
        if ("TestInt2".Equals(key)) return _TestInt2;
        if ("TestEnumName".Equals(key)) return _TestEnumName;
        return null;
    }
    
    public static DataSpawn Read(string fileName, IScorpioReader reader) {
        var ret = new DataSpawn();
        ret._ID = reader.ReadInt32();
        ret._TestInt = reader.ReadInt32();
        ret._TestString = reader.ReadString();
        ret._TestLanguage = reader.ReadString();
        ret._TestBool = reader.ReadBool();
        ret._TestInt2 = Int2.Read(fileName, reader);
        ret._TestEnumName = (TestEnum)reader.ReadInt32();
        return ret;
    }
    
    public void Set(DataSpawn value) {
        this._ID = value._ID;
        this._TestInt = value._TestInt;
        this._TestString = value._TestString;
        this._TestLanguage = value._TestLanguage;
        this._TestBool = value._TestBool;
        this._TestInt2 = value._TestInt2;
        this._TestEnumName = value._TestEnumName;
    }
    
    public override string ToString() {
        return $"ID:{_ID}, TestInt:{_TestInt}, TestString:{_TestString}, TestLanguage:{_TestLanguage}, TestBool:{_TestBool}, TestInt2:{_TestInt2}, TestEnumName:{_TestEnumName}, ";
    }
}
}

File: Backup/Go/src/scov/DataSpawn.go
package scov
import (
    "scorpioproto"
)

// DataSpawn 类名
type DataSpawn struct {
    
    _ID int32
    _TestInt int32
    _TestString string
    _TestLanguage string
    _TestBool bool
    _TestInt2 *Int2
    _TestEnumName TestEnum
}


// ID 测试ID 此值必须唯一 而且必须为int型  默认值()
func (data *DataSpawn) ID() int32 { return data._ID; }

// GetID 测试ID 此值必须唯一 而且必须为int型  默认值()
func (data *DataSpawn) GetID() int32 { return data._ID; }

// GetTestInt int类型  默认值()
func (data *DataSpawn) GetTestInt() int32 { return data._TestInt; }

// GetTestString string类型  默认值()
func (data *DataSpawn) GetTestString() string { return data._TestString; }

// GetTestLanguage 测试多国语言  默认值()
func (data *DataSpawn) GetTestLanguage() string { return data._TestLanguage; }

// GetTestBool bool类型  默认值()
func (data *DataSpawn) GetTestBool() bool { return data._TestBool; }

// GetTestInt2 自定义类型 根据ExcelConfig下 table.sco文件定义的Int2解析 类型为table_后面的名字 格式为 , 隔开  默认值()
func (data *DataSpawn) GetTestInt2() *Int2 { return data._TestInt2; }

// GetTestEnumName 自定义枚举  默认值()
func (data *DataSpawn) GetTestEnumName() TestEnum { return data._TestEnumName; }


// GetData 获取数据
func (data *DataSpawn) GetData(key string) interface{} {
    if "ID" == key {
        return data._ID;
    }
    if "TestInt" == key {
        return data._TestInt;
    }
    if "TestString" == key {
        return data._TestString;
    }
    if "TestLanguage" == key {
        return data._TestLanguage;
    }
    if "TestBool" == key {
        return data._TestBool;
    }
    if "TestInt2" == key {
        return data._TestInt2;
    }
    if "TestEnumName" == key {
        return data._TestEnumName;
    }
    return nil;
}


// Set 设置数据
func (data *DataSpawn) Set(value *DataSpawn) {
    data._ID = value._ID;
    data._TestInt = value._TestInt;
    data._TestString = value._TestString;
    data._TestLanguage = value._TestLanguage;
    data._TestBool = value._TestBool;
    data._TestInt2 = value._TestInt2;
    data._TestEnumName = value._TestEnumName;
}


// Read 读取数据
func (data *DataSpawn) Read(fileName string, reader scorpioproto.IScorpioReader) {
    data._ID = reader.ReadInt32();
    data._TestInt = reader.ReadInt32();
    data._TestString = reader.ReadString();
    data._TestLanguage = reader.ReadString();
    data._TestBool = reader.ReadBool();
    data._TestInt2 = Int2Read(fileName, reader);
    data._TestEnumName = TestEnum(reader.ReadInt32());
}

// DataSpawnRead 读取数据
func DataSpawnRead(fileName string, reader scorpioproto.IScorpioReader) *DataSpawn {
    ret := new(DataSpawn)
    ret.Read(fileName, reader)
    return ret
}
